u2f-chrome-extension/jasmine/spec/SoftTokenHelperSpec.js

```diff
@@ -0,0 +1,365 @@
+    });
+
+    it("should send WRONG_DATA_STATUS if we already own one of the keyhandles",
+       function() {
+         softTokenProfile.registrations = [exampleRegistration];
+         var softTokenEnrollResponse = softTokenEnrollHandler.run(this.callback);
+         expect(softTokenEnrollResponse).toBe(true);
+         expect(this.callback)
+           .toHaveBeenCalledWith(expectedEnrollResponseAlreadyMatchingKeyHandle);
+    });
+  });
+
+  describe("Signing", function() {
+    it("should fail if no key exists", function() {
+      var softTokenSignResponse = softTokenSignHandler.run(this.callback);
+      expect(softTokenSignResponse).toBe(true);
+      expect(this.callback)
+        .toHaveBeenCalledWith(expectedSoftTokenSignResponseNoMatchingKey);
+    });
+
+    it("should fail if a different key exists", function() {
+      var softTokenEnrollResponse = softTokenEnrollHandler.run(this.callback);
+      expect(softTokenEnrollResponse).toBe(true);
+
+      // Check a different key
+      var softTokenSignResponse = softTokenSignHandler.run(this.callback);
+      expect(softTokenSignResponse).toBe(true);
+      expect(this.callback)
+        .toHaveBeenCalledWith(expectedSoftTokenSignResponseNoMatchingKey);
+    });
+
+    it("should succeed if a matching key does exist", function() {
+      softTokenProfile.registrations = [exampleRegistration];
+      var softTokenSignResponse = softTokenSignHandler.run(this.callback);
+      expect(softTokenSignResponse).toBe(true);
+      expect(this.callback)
+        .toHaveBeenCalledWith(expectedSoftTokenSignResponseMatchingKey);
+    });
+  });
+
+  describe("TransferAccess", function() {
+    var signHelperRequestForTransferAccess;
+    var softTokenSignHandlerForTransferAccess;
+    var expectedSoftTokenSignResponseTransferAccess;
+    var expectedRegistration;
+
+    beforeEach(function() {
+      signHelperRequestForTransferAccess = {
+        type: "sign_helper_request",
+        timeoutSeconds: timeoutSeconds,
+        signData: [
+          {
+            version: version,
+            appIdHash: appIdHash,
+            challengeHash: challengeHash,
+            keyHandle: originalKeyHandle
+          }
+        ]
+      };
+
+      softTokenSignHandlerForTransferAccess =
+        new SoftTokenSignHandler(softTokenProfile,
+                                 signHelperRequestForTransferAccess);
+
+      expectedSoftTokenSignResponseTransferAccess =
+        exampleValidTransferAccessMessage;
+
+      expectedRegistration = {
+        appIdHash: appIdHash,
+        keyHandle: keyHandle,
+        keys: {
+          sec: jasmine.any(String),
+          pub: pubKey
+        },
+        counter: 1
+      };
+    });
+
+    // softokkenhelper to softokkenhelper communication
+    it("should generate key handle and public key when asked", function() {
+    });
+
+    it(
+      "should return an attestation cert and <key handle, pubkey> pair in \
```

Comment:
These are testing skeletons for the token to token communication portion of the transfer access protocol. I don't know if we'll implement this (per our conversation) so they are currently empty.